Tests/Unit/Service/DumperTest.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace Presta\SitemapBundle\Tests\Unit\Service;
+
+use PHPUnit\Framework\TestCase;
+use Presta\SitemapBundle\Event\SitemapPopulateEvent;
+use Presta\SitemapBundle\Service\Dumper;
+use Presta\SitemapBundle\Sitemap\Url\UrlConcrete;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\Filesystem\Filesystem;
+
+class DumperTest extends TestCase
+{
+    private const DUMP_DIR = __DIR__ . '/.artifacts';
+
+    /**
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var Dumper
+     */
+    private $dumper;
+
+    private static function createDir(): void
```

Comment:
I guess you didn't take advantage of the Symfony `Filesystem` features here to reduce dependencies in the unit tests?

Reply:
You're right. I've no strong opinion on this particular point.
Maybe it's a bit overkill... What do you think?

Reply:
Your code is quite simple and it's easy to understand.
It's obviously more robust to reduce the dependencies, especially when supporting several versions of Symfony.
One could assume that the component behavior may change from one version to another.

But because the Symfony `Filesystem` component is already a dependency of the project, because it's actually really stable and because I don't like to "reinvent the wheel", if I had been writing this test, I would have used it.